src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetSpelling.java

```diff
@@ -105,6 +107,7 @@ public TextEditingTargetSpelling(DocDisplay docDisplay,
                   "spelling"));
             }
          }
+         typoSpellChecker_.prefetchWords(prefetchWords);
```

Comment:
This is going to need a lot of throttling so that opening up a document with a lot of misspelled words doesn't peg your CPU for mysterious reasons as thread(s) spin up. At a minimum I think we'll need to put a conservative upper limit on the number of words we prefetch (perhaps weighted by length since that's how the algorithm scales), and possibly also limit prefetching to visible words.

Reply:
(It's okay if we plan to add that throttling later, but it should be on the checklist!)

Reply:
Yeah, was going to put a comment here but forgot to about how this is not how it'll be in the final product . The list of optimizations that we talked about last week are in-flight/coming up right behind this PR, I just wanted to keep this guy separated from more piecemeal stuff. Lots of smaller optmizations with only prefetching what's in the current viewport and only prefetching X in Y time etc.